python/lang/security/audit/dangerous-asyncio-shell-tainted-env-args.yaml

```diff
@@ -0,0 +1,88 @@
+rules:
+  - id: dangerous-asyncio-shell-tainted-env-args
+    mode: taint
+    options:
+      symbolic_propagation: true
+    pattern-sources:
+      - patterns:
+          - pattern-either:
+              - patterns:
+                  - pattern-either:
+                      - pattern: os.environ['$ANYTHING']
+                      - pattern: os.environ.get('$FOO', ...)
+                      - pattern: os.environb['$ANYTHING']
+                      - pattern: os.environb.get('$FOO', ...)
+                      - pattern: os.getenv('$ANYTHING', ...)
+                      - pattern: os.getenvb('$ANYTHING', ...)
+              - patterns:
+                  - pattern-either:
+                      - patterns:
+                          - pattern-either:
+                              - pattern: sys.argv[...]
+                              - pattern: sys.orig_argv[...]
+                      - patterns:
+                          - pattern-inside: |
+                              $PARSER = argparse.ArgumentParser(...)
+                              ...
+                          - pattern-inside: |
+                              $ARGS = $PARSER.parse_args()
+                          - pattern: <... $ARGS ...>
+                      - patterns:
+                          - pattern-inside: |
+                              $PARSER = optparse.OptionParser(...)
+                              ...
+                          - pattern-inside: |
+                              $ARGS = $PARSER.parse_args()
+                          - pattern: <... $ARGS ...>
+                      - patterns:
+                          - pattern-either:
+                              - pattern-inside: |
+                                  $OPTS, $ARGS = getopt.getopt(...)
+                                  ...
+                              - pattern-inside: |
+                                  $OPTS, $ARGS = getopt.gnu_getopt(...)
+                                  ...
+                          - pattern-either:
+                              - patterns:
+                                  - pattern-inside: |
+                                      for $O, $A in $OPTS:
+                                        ...
+                                  - pattern: $A
+                              - pattern: $ARGS
+    pattern-sinks:
+      - patterns:
+          - pattern-either:
+              - pattern-inside: $LOOP.subprocess_shell($PROTOCOL, $CMD)
+              - pattern-inside: asyncio.subprocess.create_subprocess_shell($CMD, ...)
+              - pattern-inside: asyncio.create_subprocess_shell($CMD, ...)
+          - pattern: $CMD
```

Comment:
same focus-metavariable?

Reply:
Addressed in a7df6ca8bd66541db46740c0f5c62e6b8f5bc648